sqlglot/generator.py

```diff
@@ -218,6 +219,15 @@ def sep(self, sep=" "):
     def seg(self, sql, sep=" "):
         return f"{self.sep(sep)}{sql}"
 
+    def maybe_comment(self, sql, comment):
+        if not comment or not self._comments:
+            return sql
+
+        if not self.pretty or "\n" not in comment:
+            return f"{sql} /* {comment.strip()} */"
```

Comment:
looks like we don't do -- ? i thought we'd have the ability to do -- if pretty and no line breaks

Reply:
Yes -- I've documented why using `--` is incorrect with the current formatting rules in the generator [here](https://github.com/tobymao/sqlglot/pull/663#issuecomment-1301583400).

Reply:
Discussed offline, maybe it's possible we can fix this. I'll take a look.